pkg/queue/manager_test.go

```diff
@@ -195,6 +196,40 @@ func TestUpdateClusterQueue(t *testing.T) {
 	}
 }
 
+func TestCohortCycle(t *testing.T) {
+	cohorts := []*kueuealpha.Cohort{
+		utiltesting.MakeCohort("cohort-a").Parent("cohort-b").Obj(),
+		utiltesting.MakeCohort("cohort-b").Parent("cohort-c").Obj(),
+		utiltesting.MakeCohort("cohort-c").Parent("cohort-a").Obj(),
+	}
+	cq := utiltesting.MakeClusterQueue("cq1").Cohort("cohort-a").Obj()
+	lq := utiltesting.MakeLocalQueue("foo", defaultNamespace).ClusterQueue("cq1").Obj()
+	wl := utiltesting.MakeWorkload("a", defaultNamespace).Queue("foo").Creation(time.Now()).Obj()
+	// Setup.
+	ctx := context.Background()
+	cl := utiltesting.NewFakeClient(
+		&corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: defaultNamespace}},
+	)
+	manager := NewManager(cl, nil)
+	for _, cohort := range cohorts {
+		manager.AddOrUpdateCohort(ctx, cohort)
+	}
+	if err := manager.AddClusterQueue(ctx, cq); err != nil {
+		t.Fatalf("Failed adding clusterQueue %s: %v", cq.Name, err)
+	}
+	if err := manager.AddLocalQueue(ctx, lq); err != nil {
+		t.Fatalf("Failed adding queue %s: %v", lq.Name, err)
+	}
+	if err := cl.Create(ctx, wl); err != nil {
+		t.Fatalf("Failed adding workload to client: %v", err)
+	}
+	manager.RequeueWorkload(ctx, workload.NewInfo(wl), RequeueReasonGeneric)
+
+	if manager.moveWorkloadsCohort(ctx, manager.hm.Cohorts["cohort-a"]) {
```

Comment:
Can you add some context to this part of the test to emphasize this method checks for cycle?

Reply:
Done.